### PR TITLE
Don't timestamp active log file

### LINKED ISSUE #11061: Logfiles are always generated with a '-{timestamp}.log' suffix?
According to the docs ([here](https://www.nomadproject.io/docs/configuration#log_file) and [here](https://www.nomadproject.io/docs/configuration/audit#path)) it should be possible to configure Nomad to output agent logs and audit logs to a **specific** file. This seems to be _slightly_ broken as I only ever see logs being generated with a  `-{timestamp}.log` suffix ?

It **does** respect the file name (see the examples below) ... but it always adds the `-{timestamp}.log` suffix even though this should, according to the [docs](https://www.nomadproject.io/docs/configuration/audit#path),  only happen upon file rotation.

Nomad version is `v1.1.2+ent`.

Configuration of the ordinary agent logs in nomad.hcl :point_down: 

```
# log settings
...
log_file  = "/var/log/nomad.log"
...
```

Configuration of the Enterprise audit logs in nomad.hcl :point_down: 

```  
sink "audit" {
    ...
    path   = "/var/log/nomad_audit.log"
    ...
}
```

The result of this config ...

```
root@xxx:/var/log# ls -alh /var/log/nomad*
-rw-r----- 1 root root 8.3K Aug 18 11:22 /var/log/nomad-1629283904922740456.log
-rw-r----- 1 root root 7.7K Aug 18 11:22 /var/log/nomad-1629285757963036138.log
-rw------- 1 root root 2.1M Aug 18 11:22 /var/log/nomad_audit-1629283907665719584.log
-rw------- 1 root root 1.1M Aug 18 11:42 /var/log/nomad_audit-1629285759194447584.log

```

Likewise, trying to change the file permissions for the audit log file (per these [instructions](https://www.nomadproject.io/docs/configuration/audit#mode)) does not seem to work as the files are **always** created with `0600` - I am not sure if the 2 issues are related but I thought I might as well mention it here.

### PR DESCRIPTION
**Pull request by lgfa29**
The `log_file` configuration option allows setting the log file name to use. In practice, however, this value was being used as a filename prefix with a timestamp appended.

This PR changes the log file creation logic to always use the value set in `log_file` as the active log file name.

Closes #11061 #9582

### CODE REVIEW

File: website/content/docs/upgrade/upgrade-specific.mdx [hunk 1]
```diff
@@ -13,6 +13,15 @@ upgrade. However, specific versions of Nomad may have more details provided for
...
 used to document those details separately from the standard upgrade flow.
 
+## Nomad 1.1.4 and 1.2.0
+
+#### Log file names
+
+The [`log_file`] configuration option was not being fully respected, as the
+generated filename would include a timestamp. After upgrade, the active log
+file will always be the value defined in `log_file`, with timestamped files
+being created during log rotation.
```
Comment (schmichael) [inline website/content/docs/upgrade/upgrade-specific.mdx:23]: > ...with timestamped files being created during log rotation.

Maybe be even more explicit:

> Timestamped files will only be created if \[log rotation\]\(...\) is configured.

**Review by schmichael — APPROVED** [review]
Thanks @henrikjohansen for reporting the issue and @lgfa29 for fixing it!

File: website/content/docs/upgrade/upgrade-specific.mdx (hunk 1 above)
Comment (lgfa29) [reply website/content/docs/upgrade/upgrade-specific.mdx:23]: I think log rotation is always enabled. The controls you have are `log_rotate_bytes` and `log_rotate_duration`, which defaults to `24h`. 

You can set `log_rotate_bytes` to `0` to not rotate on size. If you set `log_rotate_duration` to `0` you will have a bad time 😬

Comment (schmichael) [reply website/content/docs/upgrade/upgrade-specific.mdx:23]: Yet another reason why I need to run more long lived clusters. Thanks! :sweat_smile:

**Pull request merged**